### PR TITLE
Rename custom_metrics to extra_metrics

### PR DESCRIPTION
**Pull request by annzhang-db**
### Related Issues/PRs

 #xxx

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/models/evaluation/default_evaluator.py [hunk 1]
```diff
@@ -1549,13 +1550,27 @@ def evaluate(
...
         self.feature_names = dataset.feature_names
-        self.custom_metrics = custom_metrics
+
         self.custom_artifacts = custom_artifacts
         self.y = dataset.labels_data
         self.col_mapping = self.evaluator_config.get("col_mapping")
         self.pos_label = self.evaluator_config.get("pos_label")
         self.sample_weights = self.evaluator_config.get("sample_weights")
 
+        if extra_metrics and custom_metrics:
```
Comment (annzhang-db) [inline mlflow/models/evaluation/default_evaluator.py:1560]: check to use custom_metrics (old/deprecated) or extra_metrics and issue error or warning

File: mlflow/models/evaluation/default_evaluator.py [hunk 2]
```diff
@@ -1549,13 +1550,27 @@ def evaluate(
...
+            raise MlflowException(
+                "The 'custom_metrics' parameter in mlflow.evaluate is deprecated. Please update "
+                "your code to only use the 'extra_metrics' parameter instead."
+            )
+        if custom_metrics:
+            _logger.warning(
+                "The 'custom_metrics' parameter in mlflow.evaluate is deprecated. "
+                "Please update your code to use the 'extra_metrics' parameter instead."
+            )
```
Comment (harupy) [inline mlflow/models/evaluation/default_evaluator.py:1569]: /ai Should we use `FutureWarning` here? If yes, explain a reason why it's preferred.

Comment (harupy) [inline mlflow/models/evaluation/default_evaluator.py:1569]: Can we use `FutureWarning` here?

```python
import logging
import warnings

logger = logging.getLogger(__name__)


def future_warning():
    warnings.warn("This features is deprecated", FutureWarning, stacklevel=2)


def logger_warning():
    logger.warning("This feature is deprcated")


for _ in range(3):
    future_warning()
    logger_warning()
```

```
a.py:16: FutureWarning: This features is deprecated
  future_warning()
This feature is deprcated
This feature is deprcated
This feature is deprcated
```

`_logger.warning` prints out the message every time `evaluate` is called with `custom_metrics`, while `FutureWarning` is shown only once.

Comment (harupy) [reply mlflow/models/evaluation/default_evaluator.py:1569]: `FutureWarning` is easier to test. We can use `pytest.warns`.

**Review by sunishsheth2009 — APPROVED** [review]

**Pull request merged**